src/logging.rs

```diff
@@ -44,26 +59,112 @@ pub fn setup_layers(
         .join(","),
     )?;
 
-    let emit_kvp_layer = match EmitKVPLayer::new(
-        std::path::PathBuf::from("/var/lib/hyperv/.kvp_pool_1"),
-        vm_id,
-    ) {
-        Ok(layer) => Some(layer.with_filter(kvp_filter)),
-        Err(e) => {
-            event!(Level::ERROR, "Failed to initialize EmitKVPLayer: {}. Continuing without KVP logging.", e);
-            None
+    let kvp_enabled = config
+        .map(|cfg| cfg.telemetry.kvp_diagnostics)
+        .unwrap_or(true);
+
+    let emit_kvp_layer = if kvp_enabled {
+        match EmitKVPLayer::new(
+            std::path::PathBuf::from("/var/lib/hyperv/.kvp_pool_1"),
+            vm_id,
+        ) {
+            Ok(layer) => Some(layer.with_filter(kvp_filter)),
+            Err(e) => {
+                event!(Level::ERROR, "Failed to initialize EmitKVPLayer: {}. Continuing without KVP logging.", e);
+                None
+            }
         }
+    } else {
+        event!(
+            Level::INFO,
+            "KVP diagnostics are disabled via config. Skipping KVP logging."
+        );
+        None
     };
 
     let stderr_layer = fmt::layer()
         .with_span_events(FmtSpan::NEW | FmtSpan::CLOSE)
         .with_writer(std::io::stderr)
         .with_filter(EnvFilter::from_env("AZURE_INIT_LOG"));
 
+    let log_path = config
+        .map(|cfg| cfg.azure_init_log_path.path.clone())
+        .unwrap_or_else(|| PathBuf::from(DEFAULT_AZURE_INIT_LOG_PATH));
+
+    let file_layer = match OpenOptions::new()
+        .create(true)
+        .append(true)
+        .open(&log_path)
+    {
+        Ok(file) => {
+            if let Err(e) = file.set_permissions(Permissions::from_mode(0o600))
+            {
+                event!(
+                    Level::WARN,
+                    "Failed to set permissions on {}: {}.",
+                    log_path.display(),
+                    e,
+                );
+            }
+
+            Some(
+                fmt::layer()
+                    .with_span_events(FmtSpan::NEW | FmtSpan::CLOSE)
+                    .with_writer(file)
+                    .with_filter(EnvFilter::from_env("AZURE_INIT_LOG")),
+            )
+        }
+        Err(e) => {
+            event!(
```

Comment:
I think it would be best to just not log to the file in this case.  Maybe just make sure we emit important events via syslog?

Reply:
Or at least this failure into syslog.